app/models/comment.rb

```diff
@@ -0,0 +1,6 @@
+class Comment < ActiveRecord::Base
+
+  belongs_to :user, foreign_key: 'user_id', class_name: "User"
+  belongs_to :share, foreign_key: 'share_id', class_name: "Share"
```

Comment:
We don't need to specify the `foreign_key` or `class_name` since they would be the default.  But this is ok too.